samples/kaspresso-sample/src/sharedTest/kotlin/com/kaspersky/kaspressample/sharedtest/FailingKautomatorSharedTest.kt

```diff
@@ -27,31 +26,30 @@ class FailingKautomatorSharedTest : TestCase(Kaspresso.Builder.simple(sharedTest
     val activityTestRule = ActivityTestRule(MainActivity::class.java, true, false)
 
     @Test
-    fun errorSampleTest() {
-        try {
-            before {
-                activityTestRule.launchActivity(null)
-            }.after {
-            }.run {
+    fun instrumentalTest() = kautomatorSampleTest()
 
-                step("Open Continuously Screen") {
-                    UiMainScreen {
-                        continuouslyButton {
-                            click()
-                        }
+    @Test(expected = KautomatorInUnitTestException::class)
+    fun unitTest() = kautomatorSampleTest()
+
+    private fun kautomatorSampleTest() =
+        before {
+            activityTestRule.launchActivity(null)
+        }.after {
+            // no-op
+        }.run {
+            step("Open Continuously Screen") {
+                UiMainScreen {
+                    continuouslyButton {
+                        click()
                     }
                 }
-                step("Push start button") {
-                    UiContinuouslyScreen {
-                        startButton {
-                            click()
-                        }
+            }
+            step("Push start button") {
```

Comment:
I realized that when running tests that use a **counter or timer to show a view**, they do fail on the JVM. In this case, the `FlakySafeViewBehaviorInterceptor` does not seem to help. I think this is something on the Robolectric side, although they [improved looper simulation](http://robolectric.org/blog/2019/06/04/paused-looper/) for this kind of stuff, but I didn't manage to get it working :(
I believe I deleted this step because of it. It could be that with the new approach

Reply:
I believe it was something I missed when testing it, but I'm glad it really works :)